alacritty_terminal/src/term/mod.rs

```diff
@@ -47,6 +47,9 @@ pub mod color;
 /// Used to match equal brackets, when performing a bracket-pair selection.
 const BRACKET_PAIRS: [(char, char); 4] = [('(', ')'), ('[', ']'), ('{', '}'), ('<', '>')];
 
+/// Max size of the window title stack
+const TITLE_STACK_MAX_DEPTH: usize = 8;
```

Comment:
Why do we need a max depth?

Reply:
I saw it like this with Gnome Terminal (8) and Kitty (10). This conversation in the Gnome Terminal bug report explains it: https://bugzilla.gnome.org/show_bug.cgi?id=699819#c8

Reply:
> should clear the stack on reset

That's a good point. We should do this too.

I don't think the arguments made in the gnome bugtracker make a lot of sense. They seem to assume that a bad app (talking about faulty for implementation) would only cause troubles the moment the stack is filled, but the process of the application pushing a ton of titles on the stack already messes it up completely anyways, so there's really no point I think.

Reply:
I see what you mean. I'm happy to remove that logic if you find it unnecessary. I'll also clear the stack on reset. It would be nice if we could restore the original title as well, but that would require the terminal having the original saved in it's state somewhere.

Reply:
This makes sense to me, in fact one of the first things I tested was this:

```fish
while true; echo -en "\033[22;0t"; end
```

:tada:

Reply:
> It would be nice if we could restore the original title as well, but that would require the terminal having the original saved in it's state somewhere.

I think on reset we can just reset to `Alacritty` honestly.